gui: rename debug window

Zero-1729 opened the pull request:
**Edit**: I have now limited the change in this PR to only renaming the window title from `Debug Window` to `Node Window`. Check [this comment](https://github.com/bitcoin/bitcoin/pull/17096#issuecomment-542837511) for more details.

This PR is in response to #17082, which aims to rename the `Debug window` title to a more user friendly term; `Node window`.

Closes #17082

hebasto commented:
Concept ACK.

hebasto commented:
Could scripted diff be used for renaming commits?

MarcoFalke commented on src/qt/rpcconsole.cpp:
```diff
@@ -805,7 +805,7 @@ void RPCConsole::clear(bool clearHistory)
 #endif
 
-    message(CMD_REPLY, (tr("Welcome to the %1 RPC console.").arg(PACKAGE_NAME) + "<br>" +
+    message(CMD_REPLY, (tr("Welcome to the %1 Node console.").arg(PACKAGE_NAME) + "<br>" +
```
in commit 6cdb9560dc8d301fb96cdb90c2b7c3d6fd0e4399:

I think the console is still all about RPC, so could keep the previous name?

Zero-1729 replied:
I assumed changing the message from 'RPC console' to 'Node console' would make the new window name ('Node window') consistent with the welcome message.

promag replied:
The "Node window" has 4 tabs, the RPC console is one of those.

Zero-1729 replied:
Let me change that

promag reviewed (commented):
Concept ACK, agree with the scripted diff.

Zero-1729 commented:
Sorry, the ci tests failed because I seemed to have left `'debugwindow.ui` instead of changing it to `nodewindow.ui` in `src/Makefile.qt.include`.

Zero-1729 commented:
Are there any more changes/edits I need to make?

laanwj commented:
I think it makes sense to separate the change in user facing name (in a separate PR) from the internal refactoring to change the class name. These can be discussed / bike shedded seperately.

promag commented:
Agree with @laanwj. Also, maybe we can make the current tab the window title and so we don't have to decide for a new name?

Zero-1729 commented:
> Agree with @laanwj. Also, maybe we can make the current tab the window title and so we don't have to decide for a new name?

@promag would that be changing the window title of `NodeWidget` (or `RPCConsole`) whenever a tab is set? Explicitly setting the title to tab text, i.e;

```cpp
...
void NodeWidget::on_tabWidget_currentChanged(int index)
{
    if (ui->tabWidget->widget(index) == ui->tab_console) {
        ui->lineEdit->setFocus();
    }

    // <- rename window title here
}
...
```

Zero-1729 commented:
The changes made in this PR are too much, so I have squashed them all to only include the UI facing changes (the original `gui: rename debug window` commit), as per @laanwj's suggestion. Limiting this PR to the user facing change(s) only (i.e. the renaming of the window title from `Debug window` to `Node window`). That way this would remain a simple PR, instead of a super PR.

Also, with respect to @promag's suggestion, about making the window title the current tab name, I have created a [separate branch](https://github.com/Zero-1729/bitcoin/tree/rename-debug-window-to-tab-name) with the changes included, alongside the internal class change from `RPCConsole` to `NodeWidget`, file renames, etc. That way, assuming those changes are requested, a new PR can easily be opened.

jonasschnelli reviewed (approved):
utACK b5d6aceaae878cf1b0abaf039ceb82c0e65ec246 - trivial change. Only user facing name change.

Zero-1729 commented:
@jonasschnelli  I just cloned [the branch](https://github.com/Zero-1729/bitcoin/tree/rename-debug-window) and built it from source, all the tests in `src/test/test_bitcoin-qt` were successful. 👍

The window title properly displays "Node window".

Zero-1729 commented:
Are there any other checks or changes that need to be done?

promag commented on src/qt/bitcoingui.cpp:
```diff
@@ -318,8 +318,8 @@ void BitcoinGUI::createActions()
 
-    openRPCConsoleAction = new QAction(tr("&Debug window"), this);
-    openRPCConsoleAction->setStatusTip(tr("Open debugging and diagnostic console"));
+    openRPCConsoleAction = new QAction(tr("&Node window"), this);
```
This needs to be updated:
https://github.com/bitcoin/bitcoin/blob/69a6f1ad1f7ca52b2524ab8322bfa89a9b0ee61b/src/qt/bitcoingui.cpp#L415

But I'm not sure about this shortcut, there's now a shortcut for each tab so maybe drop these?

promag replied:
Also `&N` mnemonic is associated with "Network Traffic" which is a tab in the node window.

Zero-1729 replied:
> This needs to be updated:
> https://github.com/bitcoin/bitcoin/blob/69a6f1ad1f7ca52b2524ab8322bfa89a9b0ee61b/src/qt/bitcoingui.cpp#L415
> 
> But I'm not sure about this shortcut, there's now a shortcut for each tab so maybe drop these?

So taking out the entire line since [these](https://github.com/bitcoin/bitcoin/commit/091747b46ecf06244ce2650028f1833b2e7c5062) [commits](https://github.com/bitcoin/bitcoin/commit/8019b6b150ff7444195a238470414c9deec5bf74) make the shortcut unnecessary?

Zero-1729 replied:
> Also `&N` mnemonic is associated with "Network Traffic" which is a tab in the node window.

Do you have a suggestion as to what mnemonic to replace it with? Or would you rather it be removed completely:

```cpp
openRPCConsoleAction = new QAction(tr("Node window"), this);
```

promag replied:
I'd drop it, not sure about others though.

Zero-1729 replied:
Got it :+1:

promag commented:
> Also, with respect to @promag's suggestion, about making the window title the current tab name, I have created a [separate branch](https://github.com/Zero-1729/bitcoin/tree/rename-debug-window-to-tab-name) with the changes included, alongside the internal class change from `RPCConsole` to `NodeWidget`, file renames, etc. That way, assuming those changes are requested, a new PR can easily be opened.

Oh thanks for the consideration!

Zero-1729 commented:
@promag are there any more updates?

fanquake commented:
@Zero-1729 You can [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits).

Zero-1729 commented:
@fanquake squashed :+1:

Zero-1729 commented:
Are there any more final checks or updates?

Zero-1729 commented:
Are there any more updates I need to make to the current changes?

hebasto commented on src/qt/bitcoingui.cpp:
```diff
@@ -412,7 +412,6 @@ void BitcoinGUI::createActions()
 #endif // ENABLE_WALLET
 
     connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_C), this), &QShortcut::activated, this, &BitcoinGUI::showDebugWindowActivateConsole);
-    connect(new QShortcut(QKeySequence(Qt::CTRL + Qt::SHIFT + Qt::Key_D), this), &QShortcut::activated, this, &BitcoinGUI::showDebugWindow);
```
Removing this shortcut is not mentioned in the PR description.
Moreover, I'd prefer to keep it as is.

Zero-1729 replied:
Oh, my bad, the change was in response to an earlier review by @promag.

Just read the refs and reverted the change :+1:

Zero-1729 committed:
gui: renamed 'debug window' to 'node window'

- Renamed 'Debug window' to a more user friendly term - 'Node window' - in the window title and tray menu.

fixes #17082

hebasto reviewed (approved):
ACK 44f15cfdcfc64d5a0c36fded39e4aef49415b11b, tested on Linux Mint 19.3:

![DeepinScreenshot_select-area_20200107114953](https://user-images.githubusercontent.com/32963518/71885961-0b93e380-3144-11ea-899d-6ded241e186d.png)

![DeepinScreenshot_select-area_20200107115040](https://user-images.githubusercontent.com/32963518/71885973-10589780-3144-11ea-8975-6e44669ff013.png)

Zero-1729 commented:
I now realize I can't cancel the review request I mistakenly made to @promag, as I lack the permissions. He should just ignore the request, especially since @hebasto has **ACK**ed the changes.

Zero-1729 commented:
I now realize I can't cancel [the review request  I mistakenly made](https://github.com/bitcoin/bitcoin/pull/17096#event-2825375444) to @promag a while back, as I lack the permissions. He should just ignore the request so I don't waste his time, especially since @hebasto has **ACK**ed the changes.

theStack reviewed (approved):
ACK https://github.com/bitcoin/bitcoin/commit/44f15cfdcfc64d5a0c36fded39e4aef49415b11b, tested on Linux (Lubuntu 16.04):
![17096_1](https://user-images.githubusercontent.com/91535/72898719-e8ad2600-3d24-11ea-8996-f5b080020583.png)
![17096_2](https://user-images.githubusercontent.com/91535/72898751-f6fb4200-3d24-11ea-9cf2-fdb8acbe928c.png)

promag commented:
@Zero-1729 sorry missed your review request.

Zero-1729 commented:
@promag no worries :+1:.
